Index: modules/taxonomy/taxonomy.module
===================================================================
RCS file: /cvs/drupal/drupal/modules/taxonomy/taxonomy.module,v
retrieving revision 1.414.2.5
diff -u -p -r1.414.2.5 taxonomy.module
--- modules/taxonomy/taxonomy.module	17 Sep 2008 12:55:37 -0000	1.414.2.5
+++ modules/taxonomy/taxonomy.module	3 Jun 2010 19:34:12 -0000
@@ -1177,7 +1177,9 @@ function taxonomy_nodeapi($node, $op, $a
       return taxonomy_rss_item($node);
 
     case 'update index':
-      return taxonomy_node_update_index($node);
+      if (!empty($node->taxonomy)) {
+        return taxonomy_node_update_index($node);
+      }
   }
 }
 
